--- Mac/Makefile.in.orig	2008-07-22 01:06:33.000000000 -0600
+++ Mac/Makefile.in	2008-10-04 19:51:40.000000000 -0600
@@ -18,7 +18,7 @@
 
 # These are normally glimpsed from the previous set
 bindir=$(prefix)/bin
-PYTHONAPPSDIR=/Applications/$(PYTHONFRAMEWORK) $(VERSION)
+PYTHONAPPSDIR=@@APPLICATIONS_DIR@@/$(PYTHONFRAMEWORK) $(VERSION)
 APPINSTALLDIR=$(prefix)/Resources/Python.app
 
 # Variables for installing the "normal" unix binaries
