--- Mac/Tools/Doc/setup.py.diff	2008-03-29 09:24:25.000000000 -0600
+++ Mac/Tools/Doc/setup.py	2008-10-04 19:53:40.000000000 -0600
@@ -30,7 +30,7 @@
 
 MAJOR_VERSION='2.4'
 MINOR_VERSION='2.4.1'
-DESTDIR='/Applications/MacPython-%s/PythonIDE.app/Contents/Resources/English.lproj/PythonDocumentation' % MAJOR_VERSION
+DESTDIR='@@APPLICATIONS_DIR@@/MacPython-%s/PythonIDE.app/Contents/Resources/English.lproj/PythonDocumentation' % MAJOR_VERSION
 
 class DocBuild(build):
     def initialize_options(self):
