--- src/imaging/libimagingft/_imagingft.c.orig	2014-07-13 00:37:57.000000000 +0200
+++ src/imaging/libimagingft/_imagingft.c	2014-07-13 00:41:08.000000000 +0200
@@ -70,7 +70,13 @@
     const char* message;
 } ft_errors[] =
 
+#if defined(USE_FREETYPE_2_1)
+/* freetype 2.1 and newer */
+#include FT_ERRORS_H
+#else
+/* freetype 2.0 */
 #include <freetype/fterrors.h>
+#endif
 
 /* -------------------------------------------------------------------- */
 /* font objects */
