--- numpy/distutils/fcompiler/__init__.py.orig	2013-10-31 13:24:12.000000000 +0400
+++ numpy/distutils/fcompiler/__init__.py	2013-10-31 13:45:03.000000000 +0400
@@ -815,7 +815,7 @@
     return compiler_type
 
 # Flag to avoid rechecking for Fortran compiler every time
-failed_fcompiler = False
+failed_fcompilers = []
 
 def new_fcompiler(plat=None,
                   compiler=None,
@@ -828,7 +828,8 @@
     platform/compiler combination.
     """
     global failed_fcompiler
-    if failed_fcompiler:
+    fcompiler_key = (plat, compiler)
+    if fcompiler_key in failed_fcompilers:
         return None
 
     load_all_fcompiler_classes()
@@ -848,7 +849,7 @@
             msg = msg + " Supported compilers are: %s)" \
                   % (','.join(fcompiler_class.keys()))
         log.warn(msg)
-        failed_fcompiler = True
+        failed_fcompilers.append(fcompiler_key)
         return None
 
     compiler = klass(verbose=verbose, dry_run=dry_run, force=force)
