--- configure.orig	2011-11-22 09:08:58.000000000 +1100
+++ configure	2011-11-22 09:10:56.000000000 +1100
@@ -4428,9 +4428,7 @@
 	    STRICT_PROTO="-Wstrict-prototypes"
 	fi
         # For gcc 4.x we need to use -fwrapv so lets check if its supported
-        if "$CC" -v --help 2>/dev/null |grep -- -fwrapv > /dev/null; then
            WRAP="-fwrapv"
-        fi
 	case $ac_cv_prog_cc_g in
 	yes)
 	    if test "$Py_DEBUG" = 'true' ; then
